modules/shared/system_module_version.mk

```diff
@@ -34,5 +34,12 @@ endif
 # Bump by 1 for every updated bootloader image for a release with the same v0.x.* base.
 BOOTLOADER_VERSION ?= 301
 
-# the version of the bootloader that the system firmware requires
+# The version of the bootloader that the system firmware requires
+# NOTE: this will force the device into safe mode until this dependency is met, which is why
+# this version usually lags behind the current bootloader version, to avoid non-mandatory updates.
+ifneq (,$(filter $(PLATFORM_GEN),2))
 BOOTLOADER_DEPENDENCY = 201
+endif
+ifneq (,$(filter $(PLATFORM_GEN),3))
+BOOTLOADER_DEPENDENCY = 301
```

Comment:
We should bump this for 1.2.0-rc.1 due to https://github.com/particle-iot/device-os/pull/1777/files#diff-9de5c4685f6a4024eb7b8121590af8b6R136.

Reply:
Yes we need a better way to detect when the bootloader needs to be updated :)